### PR TITLE
Add upstart and systemd support

### PR DESCRIPTION
**Pull request by alrick87**
Summary:
- Moved all the service files to the ```brooklyn-dist/src``` directory, so they're shared between RPM and .deb solutions
- Added upstart service scripts for RPM and .deb solutions
- Unified upstart and systemd services in one package

### CODE REVIEW

File: src/service/upstart/deb/brooklyn.conf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (bostko) [inline src/service/upstart/deb/brooklyn.conf]: @alrick87 use `git mv deb-packaging/src/conf/brooklyn.conf src/service/upstart/deb/brooklyn.conf` to preserve history.

**Comment by neykov** [comment]
Looks good. Would be nice to have live tests against all supported OSes for deb & rpm packages.

**Comment by neykov** [comment]
Testing with systemd I get:
```
vagrant@vagrant-ubuntu-wily-64:~$ sudo systemctl status brooklyn
● brooklyn.service - LSB: Apache Brooklyn
   Loaded: loaded (/etc/init.d/brooklyn)
   Active: active (running) since Tue 2016-05-03 12:00:07 UTC; 2min 19s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 683 ExecStart=/etc/init.d/brooklyn start (code=exited, status=0/SUCCESS)
   Memory: 284.9M
      CPU: 12.405s
   CGroup: /system.slice/brooklyn.service
           └─684 /usr/bin/java -Dbrooklyn.location.localhost.address=127.0.0.1 -Dlogback.configurationFile=/etc/brooklyn/logback.xml -Xms256m -Xmx1g -XX:MaxPermSize=256m -cp /opt/brooklyn/conf:/opt/b...
```

Looks like systemd is managing the process through the `init.d` script. Is this the expected behaviour?

**Comment by alrick87** [comment]
@neykov Thanks for pointing this out, I totally missed that. The package copies both init.d and systemd scripts, but for some reason systemd on Ubuntu 15.10 has a priority set for init.d scripts. I guess, I will just remove init.d completely as it isn't used anywhere.

**Comment by alrick87** [comment]
@neykov Removing sysvinit script fixes the issue:
```
ubuntu@wily-deb-test:~$ sudo systemctl status brooklyn
sudo: unable to resolve host wily-deb-test
● brooklyn.service - Apache Brooklyn Service
   Loaded: loaded (/etc/systemd/system/multi-user.target.wants/brooklyn.service)
   Active: active (running) since Tue 2016-05-03 13:36:57 UTC; 335ms ago
     Docs: https://brooklyn.apache.org/documentation/index.html
 Main PID: 13079 (java)
   Memory: 36.9M
      CPU: 432ms
   CGroup: /system.slice/brooklyn.service
           └─13079 /usr/bin/java -Dbrooklyn.location.localhost.address=127.0.0.1 -Dlogback.configurationFile=/etc/brooklyn/logback.xml -Xms256m -Xmx1g -X...

May 03 13:36:57 wily-deb-test systemd[1]: Started Apache Brooklyn Service.
```

**Pull request merged**